### DISCUSSION TITLE
Act-once Operators in Toolshelf Workaround

### DISCUSSION THREAD

**Discussion by Draise14** [Bforartists/Bforartists#2138] [Ideas]
Here is a clue from a user on Discord who has been trying to make operators that act once in the toolshelf with the modal based icon buttons from the toolshelf, like we tried to do. 

He pointed out a system someone else was using to hack it. 
[https://blender.stackexchange.com/questions/139714/run-modal-operator-while-tool-is-selected](https://blender.stackexchange.com/questions/139714/run-modal-operator-while-tool-is-selected)

[https://blender.stackexchange.com/questions/137710/api-throws-context-error-when-adding-ui-tool-to-sculpt-or-particle-mode](https://blender.stackexchange.com/questions/137710/api-throws-context-error-when-adding-ui-tool-to-sculpt-or-particle-mode)

Thought it interesting, so here it is to bookmark it.

**Comment by ReinerBforartists** [comment]
Definitely interesting. And maybe we need it at one point. So thanks for sharing :)
